### ISSUE TITLE
Fix bugs steering/{{ID}}/targets

### ISSUE THREAD

**Issue by dpham692** [apache/trafficcontrol#4767]
## I'm submitting a ...

- [x] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

User can create a steering target for non steering DS.
For example:
Do a **POST /api/3.0/steering/id/targets** but using other DS id besides steering.
The API will response 200 OK
```
{
    "alerts": [
        {
            "text": "steeringtarget was created.",
            "level": "success"
        }
    ],
    "response": {
        "deliveryService": "censored",
        "deliveryServiceId": "censored",
        "target": "censored",
        "targetId": 16,
        "type": "censored",
        "typeId": 43,
        "value": -1
    }
}
```
## Expected / new behavior:

API should not let user create target for non steering ds.

**Comment by mitchell852** [comment]
yeah, i suppose creating a target for a non-steering ds technically makes no sense as it won't be used....however, it has no negative effects.
